src/ndpp_init.F90

```diff
@@ -0,0 +1,622 @@
+          if (associated(data % inel_Ein)) then
+            deallocate(data % inel_Ein)
+            data % inel_Ein => ref % inel_Ein
+          end if
+          if (associated(data % inel_Ein_srch)) then
+            deallocate(data % inel_Ein_srch)
+            data % inel_Ein_srch => ref % inel_Ein_srch
+          end if
+
+          ! Ok, good work team, now mark skip as true so we dont do it again
+          skip(i_other) = .True.
+
+          ! Clean up the pointers
+          nullify(data)
+          nullify(ref)
+        end if
+      end do
+    end do
+
+    deallocate(skip)
+
+  end subroutine remove_temperature_independent_data
+
+!===============================================================================
+! WHICH_DATA looks at the requested tallies and determines which data is needed
+! from the NDPP libraries so that memory utilization is kept to a minimum.
+!===============================================================================
+
+  subroutine which_data(scatt_type, get_scatt, get_nuscatt, get_chi_t, &
+                        get_chi_p, get_chi_d, scatt_order)
+    integer, intent(in)  :: scatt_type  ! Whether or not legendre or tabular data
+    logical, intent(out) :: get_scatt   ! Flag for whether or not to get scatt data
+    logical, intent(out) :: get_nuscatt ! Flag for whether or not to get nuscatt data
+    logical, intent(out) :: get_chi_t   ! Flag for whether or not to get total chi data
+    logical, intent(out) :: get_chi_p   ! Flag for whether or not to get prompt chi data
+    logical, intent(out) :: get_chi_d   ! Flag for whether or not to get delayed chi data
+    integer, intent(out) :: scatt_order ! Number of moments requested in tallies for scatter
+
+    type(TallyObject), pointer :: t => null()
+    integer :: i ! Tally index
+    integer :: j ! Score bin index
+    integer :: k ! User score bin index
+
+    ! Initialize the flags and orders
+    get_scatt = .false.
+    get_nuscatt = .false.
+    get_chi_t = .false.
+    get_chi_p = .false.
+    get_chi_d = .false.
+    scatt_order = 0
+
+    ! Step through each tally and score and determine which types are present
+    ! and the orders for scatt and nuscatt
+    TALLY_LOOP: do i = 1, n_tallies
+      t => tallies(i)
+      j = 0
+      SCORE_LOOP: do k = 1, t % n_user_score_bins
+        j = j + 1
+        select case (t % score_bins(j))
+          case (SCORE_NDPP_SCATT_N, SCORE_NDPP_SCATT_PN, SCORE_NDPP_SCATT_YN)
+            get_scatt = .true.
+            if (t % moment_order(j) > scatt_order) then
+              scatt_order = t % moment_order(j)
+            end if
+
+            ! Skip the others to save cycles
+            if (t % score_bins(j) == SCORE_NDPP_SCATT_PN) then
+              j = j + t % moment_order(j)
+              cycle SCORE_LOOP
+            else if (t % score_bins(j) == SCORE_NDPP_SCATT_YN) then
+              j = j + (t % moment_order(j) + 1)**2 - 1
+              cycle SCORE_LOOP
+            end if
```

Comment:
Again, I think the compute time you save is likely not worth the extra code (mostly because someone might try to modify it one day and mess it up, not because of lines-of-code statistics or something like that)